From 46eceefb3bd184354cdaf0fd3dff6f2ad979aad9 Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@arm.com>
Date: Sat, 30 Aug 2025 00:47:03 +0200
Subject: [PATCH] strace: skip a number of load-sensitive tests

Skip tests which are known to be unreliable under load, typically because they
care about timing.

Upstream-Status: Inappropriate
Signed-off-by: Ross Burton <ross.burton@arm.com>
Signed-off-by: Osama Abdelkader <osama.abdelkader@gmail.com>
---
 tests/clock_nanosleep.gen.test | 1 +
 tests/delay.test               | 1 +
 tests/nanosleep.gen.test       | 1 +
 tests/strace-r.test            | 1 +
 4 files changed, 4 insertions(+)

diff --git a/tests/clock_nanosleep.gen.test b/tests/clock_nanosleep.gen.test
index 7a6025bbd..f0e6dbb73 100755
--- a/tests/clock_nanosleep.gen.test
+++ b/tests/clock_nanosleep.gen.test
@@ -1,4 +1,5 @@
 #!/bin/sh -efu
 # Generated by ./tests/gen_tests.sh from ./tests/gen_tests.in (clock_nanosleep -e trace=clock_nanosleep,clock_gettime); do not edit.
 . "${srcdir=.}/init.sh"
+skip_ "Test not reliable under load"
 run_strace_match_diff -e trace=clock_nanosleep,clock_gettime
diff --git a/tests/delay.test b/tests/delay.test
index f74e27f1e..1510c7647 100755
--- a/tests/delay.test
+++ b/tests/delay.test
@@ -9,6 +9,7 @@
 
 . "${srcdir=.}/init.sh"
 
+skip_ "Test not reliable under load"
 while read -r denter dexit denter_us dexit_us; do
 	[ -n "$denter" ] || continue
 
diff --git a/tests/nanosleep.gen.test b/tests/nanosleep.gen.test
index 41928ded8..f8a87cb8e 100755
--- a/tests/nanosleep.gen.test
+++ b/tests/nanosleep.gen.test
@@ -1,4 +1,5 @@
 #!/bin/sh -efu
 # Generated by ./tests/gen_tests.sh from ./tests/gen_tests.in (nanosleep -a20 ); do not edit.
 . "${srcdir=.}/init.sh"
+skip_ "Test not reliable under load"
 run_strace_match_diff -a20 
diff --git a/tests/strace-r.test b/tests/strace-r.test
index 829973753..e89423485 100755
--- a/tests/strace-r.test
+++ b/tests/strace-r.test
@@ -9,6 +9,7 @@
 
 . "${srcdir=.}/init.sh"
 
+skip_ "Test not reliable under load"
 r_opt="${1:--r}"
 
 run_prog ../sleep 0
-- 
2.43.0

